suite-native/device-manager/src/components/DeviceManagerContent.tsx

```diff
@@ -1,90 +1,91 @@
+        dispatch(selectDeviceThunk(device));
+        setIsChangeDeviceRequested(false);
         setIsDeviceManagerVisible(false);
-        navigation.navigate(RootStackRoutes.ConnectDeviceStack, {
-            screen: ConnectDeviceStackRoutes.ConnectAndUnlockDevice,
-        });
+
         analytics.report({
             type: EventType.DeviceManagerClick,
-            payload: { action: 'connectDeviceButton' },
+            payload: {
+                action:
+                    device.id === PORTFOLIO_TRACKER_DEVICE_ID
+                        ? 'portfolioTracker'
+                        : 'connectDeviceButton',
+            },
         });
     };
 
-    const listedDevice = useMemo(
-        () => devices.filter(device => device.id !== selectedDeviceId),
-        [devices, selectedDeviceId],
-    );
-
     return (
-        <DeviceManagerModal>
-            {!isPortfolioTrackerDevice && <DeviceControlButtons />}
-            {!isPortfolioTrackerDevice && isPassphraseFeatureEnabled && <AddHiddenWalletButton />}
-            {A.isNotEmpty(listedDevice) && (
-                <VStack>
-                    <Text variant="callout">
-                        <Translation id="deviceManager.deviceList.sectionTitle" />
-                    </Text>
-                    {listedDevice.map(device => (
-                        <DeviceItem key={device.path} id={device.id} />
-                    ))}
-                </VStack>
-            )}
-            {isNoPhysicalDeviceConnected && (
-                <VStack>
-                    <Text variant="callout">
-                        <Translation id="deviceManager.connectDevice.sectionTitle" />
-                    </Text>
-                    <Button colorScheme="tertiaryElevation1" onPress={handleConnectDevice}>
-                        <Translation id="deviceManager.connectDevice.connectButton" />
-                    </Button>
-                </VStack>
-            )}
+        <DeviceManagerModal
+            customSwitchRightView={
+                !isPortfolioTrackerDevice && (
+                    <DevicesToggleButton
+                        deviceButtonState={isChangeDeviceRequested ? 'open' : 'closed'}
+                        onDeviceButtonTap={toggleIsChangeDeviceRequested}
+                    />
+                )
+            }
+            onClose={() => setIsChangeDeviceRequested(false)}
+        >
+            <VStack spacing="medium">
+                <DeviceList
+                    isVisible={isChangeDeviceRequested || isPortfolioTrackerDevice}
+                    onSelectDevice={handleSelectDevice}
+                />
+                {!isPortfolioTrackerDevice && (
+                    <>
+                        <WalletList onSelectDevice={handleSelectDevice} />
+                        <HStack style={applyStyle(deviceButtonsStyle)}>
+                            <DeviceInfoButton />
+                            {isPassphraseFeatureEnabled && !isDiscoveryActive && (
```

Comment:
Why the button is hidden while discovery is active? Why not to stop discovery if user want to open passhprase wallet?

Reply:
That is what I was told by @juriczech and made sense to me. 

And I am not sure if the discovery will be triggered again if some accounts were already discovered and device was remembered. How does this work? 🤔

Reply:
I think it could work since we can jump the device calls queue and then discovery should continue on the previous device, but I haven't tested this yet. Anyway, I would go with this as a first step and revisit this in separate issue with passphrase finalising issues if you're okay with that

Reply:
sure thing, let's revisit as part of passhprase https://github.com/trezor/trezor-suite/issues/12361